xds/internal/balancer/outlierdetection/subconn_wrapper.go

```diff
@@ -72,3 +102,56 @@ func (scw *subConnWrapper) uneject() {
 func (scw *subConnWrapper) String() string {
 	return fmt.Sprintf("%+v", scw.addresses)
 }
+
+func (scw *subConnWrapper) RegisterHealthListener(listener func(balancer.SubConnState)) {
+	// gRPC currently supports two mechanisms that provide a health signal for
+	// a connection: client-side health checking and outlier detection. Earlier
+	// both these mechanisms signaled unhealthiness by setting the subchannel
+	// state to TRANSIENT_FAILURE. As part of the dualstack changes to make
+	// pick_first the universal leaf policy (see A61), both these mechanisms
+	// started using the new health listener to make health signal visible to
+	// the petiole policies without affecting the underlying connectivity
+	// management of the pick_first policy
+	if !scw.healthListenerEnabled {
+		logger.Errorf("Health listener unexpectedly registered on SubConn %v.", scw)
+		return
+	}
+
+	scw.mu.Lock()
+	defer scw.mu.Unlock()
+
+	if scw.latestConnectivityState != connectivity.Ready {
+		return
+	}
+	scw.healthListener = listener
+	if listener == nil {
+		scw.SubConn.RegisterHealthListener(nil)
+	} else {
+		scw.SubConn.RegisterHealthListener(func(scs balancer.SubConnState) {
+			scw.scUpdateCh.Put(&scHealthUpdate{
+				scw:   scw,
+				state: scs,
+			})
+		})
+	}
+}
+
+func (scw *subConnWrapper) updateHealthState(scs balancer.SubConnState) {
```

Comment:
Nit: can we choose a name that more clearly indicates that this actually sends the `scs` to the registered health listener? The current name makes it look more like it is updating some health related state in the `subConnWrapper`.